web-client/src/pages/tasks/index.tsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+
+import axios from 'axios';
+import { AxiosError } from 'axios';
+
+import { useAsyncEffect } from "use-async-effect";
+
+import { Main } from "../../templates/Main";
+import { Meta } from "../../layout/Meta";
+import { AppConfig } from "../../utils/AppConfig";
+import { UserConstants } from '../session/types';
+
+type ShowTaskProps = {
+    taskId: number;
+}
+
+const ShowTaskPage = (props: ShowTaskProps) => {
+    const [title, setTitle] = React.useState('');
+    const [slug, setSlug] = React.useState('');
+    const [description, setDescription] = React.useState('');
+    const [statement, setStatement] = React.useState('');
+    const [allowedLanguages, setAllowedLanguages] = React.useState('All');
+    const [taskType, setTaskType] = React.useState('Batch');
+    const [scoreMax, setScoreMax] = React.useState(100);
+    const [checker, setChecker] = React.useState('');
+    const [timeLimit, setTimeLimit] = React.useState(2);
+    const [memoryLimit, setMemoryLimit] = React.useState(1099511627776);
+    const [compileTimeLimit, setCompileTimeLimit] = React.useState(10);
+    const [compileMemoryLimit, setCompileMemoryLimit] = React.useState(1099511627776);
+    const [submissionSizeLimit, setSubmissionSizeLimit] = React.useState(32768);
+    const [validator, setValidator] = React.useState('');
+    const [isPublicInArchive, setIsPublicOnArchive] = React.useState(false);
+    const [language, setLanguage] = React.useState('en-US');
+
+    async function getCurrentTask() {
+        const jwt = localStorage.getItem(UserConstants.JWT);
+
+        if (!jwt) {
+            return;
+        }
+
+        try {
+            const response = await axios.get(`http://localhost:4000/v1/tasks/show/${props.taskId}`
+                , {
+                    headers: {
+                        Authorization: jwt,
+                    },
+                }
+            );
+
+            const currentTask = response.data;
+            setTitle(currentTask.title);
+            setSlug(currentTask.slug);
+            setDescription(currentTask.description);
+            setStatement(currentTask.statement);
+            setAllowedLanguages(currentTask.allowedLanguages);
+            setTaskType(currentTask.taskType);
+            setScoreMax(currentTask.scoreMax);
+            setChecker(currentTask.checker);
+            setTimeLimit(currentTask.timeLimit);
+            setMemoryLimit(currentTask.memoryLimit);
+            setCompileTimeLimit(currentTask.compileTimeLimit);
+            setCompileMemoryLimit(currentTask.compileMemoryLimit);
+            setSubmissionSizeLimit(currentTask.submissionSizeLimit);
+            setValidator(currentTask.validator);
+            setIsPublicOnArchive(currentTask.isPublicInArchive);
+            setLanguage(currentTask.language);
```

Comment:
in the future, it's possible we want to add more things to the task. so it's probably better to extract the common type here and have that as a single state var. it feels like what we want is actually:
```tsx
type Task = {
  title: string;
  slug: string;
  // etc.
};

const [task, setTask] = useState<Task>(null);
// etc.
const currentTask = response.data as Task;
setTask(currentTask);
// etc.

const { title, slug, description, /* etc */ } = task;
```

that way you can also reuse the same `Task` type in `slug.tsx` as well

Reply:
it sounds like there might already be a `Task` type though, idk how whateverorm works haha. @jabbawookiees would know